webapp/channels/src/packages/mattermost-redux/src/actions/emojis.ts

```diff
@@ -69,16 +72,46 @@ export function getCustomEmojiByName(name: string): ActionFunc {
 }
 
 export function getCustomEmojisByName(names: string[]): ActionFunc {
-    return async (dispatch: DispatchFunc) => {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
         if (!names || names.length === 0) {
             return {data: true};
         }
 
-        const promises: Array<Promise<ActionResult|ActionResult[]>> = [];
-        names.forEach((name) => promises.push(dispatch(getCustomEmojiByName(name))));
+        let data;
+        try {
+            data = await Client4.getCustomEmojisByNames(names);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
 
-        await Promise.all(promises);
-        return {data: true};
+        const actions: AnyAction[] = [{
+            type: EmojiTypes.RECEIVED_CUSTOM_EMOJIS,
+            data,
+        }];
+
+        if (data.length !== names.length) {
```

Comment:
I don't understand this check: can you elaborate?

Reply:
This is just a shortcut to save us from having to loop through all the names/emojis to check if any were missed since, if that was the case, the results we received would be shorter than the number of names we requested

Reply:
Ah, this is really nuanced. Have we measure some performance overhead here that requires this kind of optimization at the expense of code clarity?

Reply:
Maybe I misunderstood what you were asking. This whole block is to keep track of which emojis we've requested which weren't returned by the server because they don't exist. Some pieces of code (mostly `getCustomEmojisInText`) can get things that look like emojis but aren't real ones, so we want to track which ones aren't actually emojis that to avoid re-requesting these ones.

Do the changes I made to address Daniel's concerns make the rest of this block's purpose clearer?

Reply:
a

Reply:
I think the other changes are orthogonal. I'm reading `data.length !== names.length` as a way to avoid running the block of code for performance reasons, as it's otherwise safe to run through them all, if a bit redundant since if we ask for 250 emoijs and get back 250 emojis, it seems unlikely that any were missed.

I suppose I'm just wondering if the check is "necessary", because we run the code below so often that it's worth avoiding it in the first place. Maybe there's some weird bug in the future that has the server returning 250 emojis, but they aren't the ones we asked for? Would this check obscure that fact and make it harder to debug?

I'm really only 1/5 here, so no blocking required -- just thinking out loud :)

Reply:
It was more necessary with the original version of this loop which was uglier and less performant, but it's half a performance thing since it saved us from doing the originally-O(n^2) inner loop and half a visual organization thing since the code block contains all the code needed for determining which emojis are missing